src/main/java/de/zalando/zmon/dataservice/DataServiceMetrics.java

```diff
@@ -152,9 +177,23 @@ public void markAccount(String account, Optional<String> region, int size) {
         }
     }
 
-    public void markCheck(int checkId, int size) {
-        getOrCreateMeter(checkMeters, "ds.check." + checkId + ".rate").mark(size);
-        getOrCreateCounter(checkCounter, "ds.check." + checkId + ".counter").inc();
+    public void markCheck(int checkId, int checkDataSize, int totalFieldsIngested, int metricsFieldsIngested, String account, Optional<String> region) {
+
+        checkArgument(!isNullOrEmpty(account));
+        checkNotNull(region);
+
+        String base;
+
+        if (region.isPresent()) {
+            base = String.format("ds.check.%s.%s.%s", checkId, account, region.get());
+        } else {
+            base = String.format("ds.check.%s.%s", checkId, account);
+        }
+
+        getOrCreateMeter(checkDataIngestedMeters, base + ".rate").mark(checkDataSize);
```

Comment:
This is already counted under another name outside the per-check check.  Once removed this would allow nesting the per-check metric into the Kairos writer directly reducing complexity.